### DISCUSSION TITLE
Why MagicFilter in aiogram doesn't have "__contains__", but have "in_"?

### DISCUSSION THREAD

**Discussion by aminabbasov** [aiogram/aiogram#1476] [Q&A] [answered]
In my opinion, having the option to check the presence of `F` or `MagicFilter` with built-in Python syntax will be much better than the current `in_` method.

My view:
```python
F.content_type in {ContentType.PHOTO, ContentType.VIDEO}
```
Current:
```python
F.content_type.in_({ContentType.PHOTO, ContentType.VIDEO})
```

But I think this is quite an obvious idea to implement `__contains__` and check element existence in a collection, so I guess that there is a reason for not doing that. And I want to know it, if so =)

**Comment by JrooTJunior** [comment, marked as answer]
Because `in` operator calls `__contains__` on the right operand inplace and returns always boolean value and the magic filter, on the contrary, works in a delayed manner when necessary (on imcoming events).

So when you execute the following:
```python
F.content_type in {ContentType.PHOTO, ContentType.VIDEO}
```
The presence of the filter object (`F.content_type` == delayed getattr) in the sequence shown on the right will be checked immediately.

and when you execute
```python
F.content_type.in_({ContentType.PHOTO, ContentType.VIDEO})
```
the `contains` operation will be added to the filter and when `resolve` method the presence of the value of the specified attribute will be checked.

---

The same as in the [SQLAlchemy `ColumnClause.in_` operator](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.in_)

**Reply by aminabbasov** [reply]
Thanks for the answer! And by chance, I want to express my gratitude for `aiogram`! I remember trying `python-telegram-bot` before discovering it, and I found it rather inconvenient. However, using `aiogram` has been a much more pleasant and easier experience. Thank you! :)
